marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/NodeClient.hs

```diff
@@ -244,7 +252,16 @@ mkClientStNext costModel maxCost changesVar getHeaderAtPoint pipelineDecision n
       let clientTip = At blockNo
       pure $ mkClientStIdle costModel maxCost changesVar getHeaderAtPoint pipelineDecision n clientTip tip
   , recvMsgRollBackward = \point tip -> do
-      clientTip <- fmap blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+      tipFromChanges <- atomically $ lookupPointInChanges point <$> readTVar changesVar
+      clientTip <- case tipFromChanges of
+        Nothing -> do
+          mClientTip <- (fmap . fmap) blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+          case mClientTip of
+            Nothing -> error $ "Failed to process rollback. Failed to load block number for rollback point " <> show point
```

Comment:
So if there is a rollback of the rollback, we might see this error?

Reply:
No, most likely not - we'd still be rolling back to a known block.